src/realm/sync/client_base.hpp

```diff
@@ -291,6 +292,11 @@ struct ClientConfig {
     ///
     /// Testing/debugging feature. Should never be enabled in production.
     bool disable_sync_to_disk = false;
+
+    /// Default implementation of the WebSocket factory. Can be overriden in SDK
+    std::function<std::unique_ptr<realm::util::websocket::SocketFactory>(
```

Comment:
I still think that having factories for factories is an anti-pattern we should avoid. Why can't SDKs just pass in an instance of a SocketFactory here? The only things we pass into the SocketFactory that could be internal to the sync client is the logger, and that's a stretch.